.github/workflows/check.yaml

```diff
@@ -25,15 +25,40 @@ jobs:
       python-version: ${{ matrix.python-version }}
 
   func:
-    uses: canonical/bootstack-actions/.github/workflows/func.yaml@v3
     needs: lint-unit
+    name: functional tests
+    runs-on: ${{ matrix.architecture.runs-on }}
+    timeout-minutes: 120
     strategy:
       fail-fast: false
-    with:
-      command: "TEST_JUJU3=1 make functional"  # using TEST_JUJU3 due https://github.com/openstack-charmers/zaza/commit/af7eea953dd5d74d3d074fe67b5765dca3911ca6
-      runs-on: "['self-hosted', 'linux', 'x64', 'large', 'jammy']"
-      juju-channel: "3.4/stable"
-      nested-containers: false
-      provider: "lxd"
-      python-version: "3.10"
-      timeout-minutes: 120
+      matrix:
+        juju-channel: ["3.4/stable"]
+        architecture:
+          - runs-on: [self-hosted, linux, x64, large, jammy]
```

Comment:
NOTE: we're using the large self-hosted instance here because we need more resources for the test zaza bundle.

Reply:
Got it, thanks

Reply:
Sorry, this was more a note-to-self.  It would be good to document it in a comment in check.yaml, but we can add it later when we port it to the automation repo. :)

Reply:
Resolving because this can be added later to automation repo